cypress/integration/interface_builder_test.js

```diff
@@ -1,3 +1,329 @@
+const _ = require('lodash');
+
+const parseInterfaceWithoutDefaults = ({ ...iface }) => {
+  if (iface.type === 'datastream' && iface.aggregation === 'individual') {
+    delete iface.aggregation;
+  }
+  if (iface.type === 'datastream') {
+    iface.mappings.forEach((mapping) => {
+      if (mapping.explicit_timestamp === false) {
+        delete mapping.explicit_timestamp;
+      }
+      if (mapping.reliability === 'unreliable') {
+        delete mapping.reliability;
+      }
+      if (mapping.retention === 'discard') {
+        delete mapping.retention;
+      }
+      if (mapping.expiry === 0) {
+        delete mapping.expiry;
+      }
+      if (mapping.database_retention_policy === 0) {
+        delete mapping.database_retention_policy;
+      }
+      if (mapping.database_retention_ttl === 60) {
+        delete mapping.database_retention_ttl;
+      }
+    });
+  }
+  if (iface.type === 'properties') {
+    iface.mappings.forEach((mapping) => {
+      if (mapping.allow_unset === false) {
+        delete mapping.allow_unset;
+      }
+    });
+  }
+  return iface;
+};
+
+const parseMappingOptions = (mapping) => {
+  return {
+    reliability: _.get(mapping, 'reliability', 'unreliable'),
+    explicitTimestamp: _.get(mapping, 'explicit_timestamp', false),
+    retention: _.get(mapping, 'retention', 'discard'),
+    expiry: _.get(mapping, 'expiry', 0),
+    databaseRetention: _.get(mapping, 'database_retention_policy', 'no_ttl'),
+    databaseTTL: _.get(mapping, 'database_retention_ttl', 60),
+    allowUnset: _.get(mapping, 'allow_unset', false),
+  };
+};
+
+const setupInterfaceEditorFromSource = (iface) => {
+  cy.get('#interfaceSource')
+    .clear()
+    .type(JSON.stringify(iface), { parseSpecialCharSequences: false });
+  cy.wait(500);
+};
+
+const setupInterfaceEditorFromUI = (iface) => {
```

Comment:
I think I did take that into account. Or did you find something odd within the tests?

Reply:
it's more of a concern of advantages vs complexity of code.
For our user case i think it's easier to have smaller tests with specific interactions. Your approach is more akin to generative testing. We might actually switch to that on a later stage of the project, also the job was done and it works, not point removing that now.
Maybe next time we should go for the lazy method first, and improve upon it as our requirements change

Reply:
I see your point, I agree for the general approach. In this case I had to extract a reusable `setupInterfaceEditorFromUI` function to avoid rewriting over and over the same test code. Arguably, having smaller tests in this case could have increased the complexity of this file.